roles/openstack-setup/tasks/images.yml

```diff
@@ -1,9 +1,19 @@
 ---
 # NOTE: Glance does not know how to proxy copy-from URLs, so we should instead
 # download the image files locally and create images using the file method.
+- name: create log of downloaded images
+  file: path=/tmp/glance_images_downloaded state=touch
+
+- name: append log of downloaded images
+  lineinfile: dest=/tmp/glance_images_downloaded line="{{ item.filename }}"
+  with_items: glance.images
+  register: images_updated
+
 - name: download images locally
   get_url: url={{ item.url }} dest=/tmp/{{ item.filename }}
   with_items: glance.images
+  when: images_updated.changed
+
```

Comment:
So maybe something like
```
- name: discover existing images
  stat:
    path: /tmp/{{ item.filename }}
  run_once: True
  register: images_downloaded
  with_items: glance.images

- name: download any necessary images
  get_url:
    url: "{{ item.0.url }}"
    dest: /tmp/{{ item.0.filename }}"
  run_once: True
  when: item.1.exists
  with_together:
    - glance.images
    - images_downloaded.results
```

Taking advantage of parallel data looping: http://docs.ansible.com/ansible/playbooks_loops.html#looping-over-parallel-sets-of-data

And run_once so that we only do it on one host.

Further, we should stop deleting them at the end of this file, otherwise it doesn't matter if we test or not, every run would re-download.